src/session/session_dhandle.c

```diff
@@ -625,9 +675,11 @@ __wt_session_get_btree_ckpt(WT_SESSION_IMPL *session, const char *uri, const cha
          *
          * For named checkpoints, we don't retry, I guess because the application ought not to try
          * to open its checkpoints while regenerating them.
+         *
+         * If the current checkpoint could not be used, check if there's more, otherwise we are
+         * done.
          */
-
-    } while (is_unnamed_ckpt && (ret == WT_NOTFOUND || ret == EBUSY));
+    } while (is_unnamed_ckpt && ckptbase->name != NULL && (ret == WT_NOTFOUND || ret == EBUSY));
```

Comment:
The test failure hints that there is a path through here where `ckptbase` could be NULL:

```
[2023/02/24 02:52:20.532] ==39388==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x7fc52c5ae748 bp 0x7fc5212ef4d0 sp 0x7fc5212ef1a0 T15)
[2023/02/24 02:52:20.532] ==39388==The signal is caused by a READ memory access.
[2023/02/24 02:52:20.532] ==39388==Hint: address points to the zero page.
[2023/02/24 02:52:20.532]     #0 0x7fc52c5ae748 in __wt_session_get_btree_ckpt /data/mci/21de56d16c165b513be488e4796a1be7/wiredtiger/src/session/session_dhandle.c:682:48
```